src/modelinstance.cpp

```diff
@@ -1461,41 +1461,51 @@ Status ModelInstance::inferAsync(const RequestType* requestProto,
+                auto status = serializePredictResponse(outputGetter, getName(), getVersion(), getOutputsInfo(), requestProto, res.get(), getTensorInfoName, useSharedOutputContentFn(requestProto));  // TODO FIXME handle status
+            } catch (std::exception& e) {
+                SPDLOG_DEBUG("caught exception in ov::InferRequest callback: {}", e.what());
             } catch (...) {
-                SPDLOG_DEBUG("got exception in ov::InferRequest callback");
-                return;
+                SPDLOG_DEBUG("caught exception in ov::InferRequest callback");
             }
-        }
-        SPDLOG_DEBUG("Using OV callback");
-        // here use OVMS response serialization
-        // here call user set callback
-        // here will go OVMS C-API serialization code
-        std::unique_ptr<ResponseType> res(new ResponseType(this->getName(), this->getVersion()));
-        OutputGetter<ov::InferRequest&> outputGetter(inferRequest);
-        try {
-            // TODO created filter based on what is in request, then perform casual serialization for what was NOT in request, and rewrite tensors from request to response for those that were
-            auto status = serializePredictResponse(outputGetter, getName(), getVersion(), getOutputsInfo(), requestProto, res.get(), getTensorInfoName, useSharedOutputContentFn(requestProto));  // TODO FIXME handle status
-        } catch (std::exception& e) {
-            SPDLOG_DEBUG("caught exception in ov::InferRequest callback: {}", e.what());
-        } catch (...) {
-            SPDLOG_DEBUG("caught exception in ov::InferRequest callback");
-        }
-        OVMS_InferenceResponse* response = reinterpret_cast<OVMS_InferenceResponse*>(res.release());
-        SPDLOG_DEBUG("Calling user provided callback");  // TODO check if this shows
-        userCallback(response, 1, userCallbackData);
-        SPDLOG_DEBUG("Called user provided callback");                       // TODO check if this shows
-        inferRequest.set_callback([](std::exception_ptr exception_ptr) {});  // reset callback on infer request // TODO this should be called on all exit paths
-    });
-    OV_LOGGER("ov::InferRequest: {}, inferRequest.start_async()", reinterpret_cast<void*>(&inferRequest));
-    inferRequest.start_async();  // TODO wrap into try catch
+            OVMS_InferenceResponse* response = reinterpret_cast<OVMS_InferenceResponse*>(res.release());
+            SPDLOG_DEBUG("Calling user provided callback");  // TODO check if this shows
+            userCallback(response, 1, userCallbackData);
+            SPDLOG_DEBUG("Called user provided callback");                       // TODO check if this shows
+            inferRequest.set_callback([](std::exception_ptr exception_ptr) {});  // reset callback on infer request // TODO this should be called on all exit paths
```

Comment:
is there task for all those todo? those seem critical in terms of release

Reply:
We have additional tasks in epic. Todos will be reviewed.